### PR TITLE
ID-1302 OD-574 [Feat] Add translation using i18n framework

### PR DESCRIPTION
**Pull request by tonytlwu**
Ref. https://weboo.atlassian.net/browse/ID-1302 https://weboo.atlassian.net/browse/OD-574

Requires https://github.com/Fliplet/fliplet-api/pull/4805

### Notes

1. I used `[html]key` to allow for HTML code within the translation code. Not sure if that's the best way to do this.
2. `T` is being highlighted as an undefined variable. I've already updated our Fliplet eslint config, which needs to be published to npm
3. I have to quit and re-run `fliplet run` every time I add new keys to `translation.json`. This is quite annoying because I would have spent time setting up the component in a certain way for testing and rerunning `fliplet run` could lose the configuration. How can we allow developers to keep editing `translation.json` without having to rerun `fliplet run`?

### CODE REVIEW

File: translation.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "loginDataSource": {
```
Comment (squallstar) [inline translation.json:2]: We should confirm this namespace across others to see how it looks like

File: translation.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "emailNotFound": "Your email address cannot be found. Please try again.",
+    "backToLogin": "Back to Login",
+    "verifyEmail": "Verify email",
+    "hasVerificationCode": "I have a verification code",
+    "verifyCodeInstruction": "A verification code was sent to <span class=\"verify-user-email\"></span>.",
+    "verificationCodePlaceholder": "Enter verification code",
+    "verificationCodeError": "You entered the wrong code or the code expired. Please try again or request a new code.",
+    "resend": "Resend verification code",
+    "verify": "Verify",
+    "resendSuccess": "A new code was sent to your email.",
```
Comment (squallstar) [inline translation.json:20]: resendCodeSuccess?

Comment (tonytlwu) [reply translation.json]: I updated the keys to use more levels of nesting. The `resendSuccess` key should have more context as well now.

**Comment by tonytlwu** [comment]
@squallstar I updated the implementation. Please see the original questions listed in the PR description.

**Comment by squallstar** [comment]
1. Is `[html]` and `[placeholder]` for defining the target of the translation? If that's how the jQuery plugin handles it (given [this](https://github.com/i18next/jquery-i18next#set-innerhtml-attributes)) then I'm good with it. 
2. You published this already
3. I think it's because Node is caching the result of `require` [here](https://github.com/Fliplet/fliplet-cli/blob/a38aafee62c37193bf21bce9bfb25a9814f44faa/fliplet-run.js#L192-L192). We probably need to change that to a file read from disk (`fs.readFileSync`) and then parse the JSON.

**Review by squallstar — APPROVED** [review]

**Comment by tonytlwu** [comment]
@squallstar Will this take you long to do?

> I think it's because Node is caching the result of `require` [here](https://github.com/Fliplet/fliplet-cli/blob/a38aafee62c37193bf21bce9bfb25a9814f44faa/fliplet-run.js#L192-L192). We probably need to change that to a file read from disk (`fs.readFileSync`) and then parse the JSON.

I think we should change this before giving this to Upplabs as it would speed up their development and testing.

File: translation.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          "confirmPasswordPlaceholder": "Confirm your new password",
+          "mismatch": "Passwords don't match. Please try again.",
+          "action": "Reset password",
+          "success": "You can now login using your email address and new password"
+        },
+        "errors": {
+          "sessionNotFound": "Login session not found",
+          "redirectMissing": "Screen redirect is not set up.",
+          "emailInvalid": "Please enter a valid email.",
+          "newPasswordMissing": "Enter a new password and confirm. Try again.",
```
Comment (OleksandrLavrynenko) [inline translation.json:40]: You need to add the line `"passwordMismatch": "Passwords don't match. Please try again.",`

File: translation.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          "emailPlaceholder": "Enter your email",
+          "passwordPlaceholder": "Enter your password",
+          "mismatch": "The details you entered don’t match our records. Please try again.",
+          "action": "Log in",
+          "forgotPassword": "Forgot your password?"
+        },
+        "restore": {
+          "instruction": "To reset your password you must verify your email address.",
+          "emailNotFound": "Your email address cannot be found. Please try again.",
+          "back": "Back to Login",
```
Comment (OleksandrLavrynenko) [inline translation.json:16]: You need to add the line `"emailMismatch": "Please enter a valid email address and try again.",`

**Review by OleksandrLavrynenko — COMMENTED** [review]
build.js line 313 => replace the text with this code `T('widgets.login.dataSource.restore.emailMismatch')`

File: translation.json (hunk 4 above)
Comment (tonytlwu) [reply translation.json:17]: @OleksandrLavrynenko Added. Thanks.

**Pull request merged**